src/components/audio-params.js

```diff
@@ -0,0 +1,271 @@
+      avatarRigObj.updateMatrixWorld(true);
+      const position = new THREE.Vector3();
+      avatarRigObj.getWorldPosition(position);
+      const direction = new THREE.Vector3(0, 0, -1);
+      const worldQuat = new THREE.Quaternion();
+      avatarRigObj.getWorldQuaternion(worldQuat);
+      direction.applyQuaternion(worldQuat);
+      return {
+        panner: {
+          orientationX: {
+            value: direction.x
+          },
+          orientationY: {
+            value: direction.y
+          },
+          orientationZ: {
+            value: direction.z
+          },
+          positionX: {
+            value: position.x
+          },
+          positionY: {
+            value: position.y
+          },
+          positionZ: {
+            value: position.z
+          },
+          distanceModel: audioParams.avatarDistanceModel,
+          maxDistance: audioParams.avatarMaxDistance,
+          refDistance: audioParams.avatarRefDistance,
+          rolloffFactor: audioParams.avatarRolloffFactor,
+          coneInnerAngle: audioParams.avatarConeInnerAngle,
+          coneOuterAngle: audioParams.avatarConeOuterAngle,
+          coneOuterGain: audioParams.avatarConeOuterGain
+        }
+      };
+    } else if (this.data.sourceType === SourceType.MEDIA_VIDEO) {
+      const audio = this.el.getObject3D("sound");
+      return audio instanceof THREE.PositionalAudio ? audio : null;
+    } else if (this.data.sourceType === SourceType.AVATAR_AUDIO_SOURCE) {
+      const audio = this.el.getObject3D("avatar-audio-source");
+      return audio instanceof THREE.PositionalAudio ? audio : null;
+    }
+  },
+
+  enableNormalizer() {
+    const audio = this.audio();
+    if (audio) {
+      const avatarAudioSource = this.el.components["avatar-audio-source"];
+      if (avatarAudioSource) {
+        this.normalizer = new AudioNormalizer(audio);
+        avatarAudioSource.el.addEventListener("sound-source-set", () => {
+          const audio = avatarAudioSource && avatarAudioSource.el.getObject3D(avatarAudioSource.attrName);
+          if (audio) {
+            this.normalizer = new AudioNormalizer(audio);
+            this.normalizer.apply();
+          }
+        });
+      }
+    }
+  },
+
+  updateDistances() {
+    const listenerPos = new THREE.Vector3();
+    this.el.sceneEl.audioListener.getWorldPosition(listenerPos);
+    this.data.distance = this.data.position.distanceTo(listenerPos);
+    this.data.squaredDistance = this.data.position.distanceToSquared(listenerPos);
+  },
+
+  updateAttenuation() {
+    if (this.data.distanceModel === "linear") {
+      this.data.attenuation = this.att_linear();
+    } else if (this.data.distanceModel === "inverse") {
+      this.data.attenuation = this.att_inverse();
+    } else if (this.data.distanceModel === "exponential") {
+      this.data.attenuation = this.att_exponential();
+    }
+  },
+
+  clipGain(gain) {
```

Comment:
Yep, it's that way just in case we would like to change the clipping gain to make it completely clip or just set it at a fixed really low gain. I guess it doesn't really make much sense...